src/ui/components/SecondaryToolbox/NewConsole.tsx

```diff
@@ -0,0 +1,388 @@
+
+// Adapter that connects the legacy app Redux stores to the newer React Context providers.
+export default function NewConsoleRoot() {
+  const recordingId = useGetRecordingId();
+
+  const sessionContext = useMemo<SessionContextType>(
+    () => ({
+      accessToken: ThreadFront.getAccessToken(),
+      recordingId,
+      sessionId: ThreadFront.sessionId!,
+
+      // Current user info is only used by the new Comments UI (which isn't included yet)
+      currentUserInfo: null as any,
+
+      // Duration info is only used by the focus editor (not imported yet)
+      duration: null as any,
+      endPoint: null as any,
+    }),
+    [recordingId]
+  );
+
+  return (
+    <SessionContext.Provider value={sessionContext}>
+      <TimelineContextAdapter>
+        <InspectorContextReduxAdapter>
+          <TerminalContextReduxAdapter>
+            <FocusContextReduxAdapter>
+              <PointsContextReduxAdapter>
+                <NewConsole showSearchInputByDefault={false} terminalInput={<JSTermWrapper />} />
+              </PointsContextReduxAdapter>
+            </FocusContextReduxAdapter>
+          </TerminalContextReduxAdapter>
+        </InspectorContextReduxAdapter>
+      </TimelineContextAdapter>
+    </SessionContext.Provider>
+  );
+}
+
+function JSTermWrapper() {
+  const [_, searchActions] = useContext(SearchContext);
+
+  const onKeyDown = (event: KeyboardEvent) => {
+    switch (event.key) {
+      case "Escape":
+        searchActions.hide();
+
+        event.preventDefault();
+        event.stopPropagation();
+        break;
+      case "f":
+      case "F":
+        if (event.metaKey) {
+          searchActions.show();
+
+          event.preventDefault();
+          event.stopPropagation();
+        }
+        break;
+    }
+  };
+
+  return (
+    <div className={styles.JSTermWrapper} onKeyDown={onKeyDown}>
+      <JSTerm />
+    </div>
+  );
+}
+
+// Adapter that reads focus region (from Redux) and passes it to the FocusContext.
+function FocusContextReduxAdapter({ children }: PropsWithChildren) {
+  const loadedRegions = useAppSelector(getLoadedRegions);
+  const focusRegion = useAppSelector(getFocusRegion);
+
+  const [isPending, startTransition] = useTransition();
+  const [deferredFocusRegion, setDeferredFocusRegion] = useState<FocusRegion | null>(null);
+
+  useEffect(() => {
+    if (focusRegion === null) {
+      return;
+    }
+
+    const updateFocusRegionOnceLoaded = () => {
+      const focusBegin = displayedBeginForFocusRegion(focusRegion);
+      const focusEnd = displayedEndForFocusRegion(focusRegion);
+      const isLoaded = loadedRegions?.loaded?.some(region => {
+        return region.begin.time <= focusBegin && region.end.time >= focusEnd;
+      });
+      if (isLoaded) {
+        startTransition(() => {
+          setDeferredFocusRegion(focusRegion);
+        });
+      } else {
+        timeoutId = setTimeout(updateFocusRegionOnceLoaded, 250);
+      }
+    };
```

Comment:
This is kind of unfortunate, but without it– the new Console might _suspend_ while asking the backend for the nearest point to a time and get an "unloaded region" error back.